### PR TITLE
fixed all async bug

### PR DESCRIPTION
**Pull request by peterBrxwn**

### CODE REVIEW

**Comment by peterBrxwn** [comment]
[[Flutter] Check native level code and make sure we always "return"](https://trello.com/c/DQgqsR0D/1570-flutter-check-native-level-code-and-make-sure-we-always-return)

**Review by arifBurakDemiray — APPROVED** [review]
Seems okey, all not-return funcs have return except throwNativeException

**Review by turtledreams — APPROVED** [review]

**Comment by turtledreams** [comment]
android side is all cool rite?

**Review by ArtursKadikis — CHANGES_REQUESTED** [review]
test all calls

**Comment by peterBrxwn** [comment]
@ArtursKadikis finished testing all calls.

File: lib/src/countly_flutter.dart [hunk 1]
```diff
@@ -503,7 +503,6 @@ class Countly {
     log('Calling "onNotification"');
     await _channel.invokeMethod('registerForNotification').then((value) {
       callback(value.toString());
-      onNotification(callback);
```
Comment (turtledreams) [inline lib/src/countly_flutter.dart:506]: lets talk about this

**Review by turtledreams — CHANGES_REQUESTED** [review]

File: lib/src/countly_flutter.dart (hunk 1 above)
Comment (turtledreams) [reply lib/src/countly_flutter.dart:506]: also test this peter with multiople push notifications and such https://countly.getoutline.com/doc/push-notifications-9qHAFIWhlc

**Review by ijunaid — APPROVED** [review]

File: lib/src/countly_flutter.dart (hunk 1 above)
Comment (turtledreams) [reply lib/src/countly_flutter.dart]: @peterBrxwn did you?

**Review by ArtursKadikis — APPROVED** [review]

**Review by turtledreams — APPROVED** [review]

**Pull request merged**